### ISSUE TITLE
get_top_gene_set.py Operon flagging bug

### ISSUE THREAD

**Issue by jackarnold13** [erinhwilson/promoter-id-from-rnaseq#1]
Hi! First off, excellent paper and tool - I really like the streamlined approach to finding promoters in non-model bacteria.

I ran into an issue when running `get_top_gene_set.py` where no matter what I put for `n%` or `min_distance` (i.e. I ran `python3 get_top_gene_set.py rg_data/rg_tpm_mat.tsv locus_tag 100 0 rg_data/sample2condition.txt rg_data/r_gnavus.gb rg_out_data/`) and every gene that was part of the analysis had a `True` boolean in the `op?` column (I suppose this could be possible, but I highly doubt that there wasn't a single gene not part of an operon...).

My best guess is that it is some issue with the formatting of my `.gb` file and how `genbank_utils.py` handles the loci position determination (Here's a[ box link ](https://uchicago.box.com/s/in34hijgrsh1ptgyzjz0ru4adgkfj06e) to the `rg_data/` directory which contains my TPM matrix `.gb` file and `sample2condition.txt` in case you wanted them for the troubleshooting as well as the `rg_out_data/` directory with the faulty file).

This wasn't a major issue for my purposes since I just changed all of the `True` values to `False` values, and just ended up with a bunch of small sequences that I filtered out since they were pretty clearly intergenic regions from an operon.

Thanks in advance, and again thank you for creating this awesome tool!

Jack Arnold
University of Chicago | PME | Mimee Lab

**Comment by erinhwilson** [comment]
Hello! Thanks for reaching out!

Ah, I think I found what was happening: in the genbank file I had been using for M. buryatense, it listed some features as type "gene" and some as type "CDS", but they were basically redundant annotations of the same thing. In the genbank feature extraction code, I put a little flag to ignore "gene" version features and instead use the "CDS" version.

In your genbank file, it appears there are 3 redundant annotations! "gene", "CDS" and "misc_feature." Because the "misc_feature" type wasn't being ignored, I think the operon filter was always finding the next upstream gene to be 0 distance away (because they were annotated as the same set of coordinates). 

![Screen Shot 2022-01-21 at 2 06 43 PM](https://user-images.githubusercontent.com/33738960/150607389-a19e1a33-4a7b-4cd5-9192-9abb95ef2e39.png)

I added a little config file called "config/gbfeats2exclude.txt" where a user can specify the types of GB features you want to ignore, mostly likely these redundant annotations or the "source" feature (aka the whole genome). I think this script should now respond to your operon filtering and not mark everything as True - let me know if it doesn't work though.

Thanks for your help finding this issue!
